### PR TITLE
Added callback which triggers when a specific IP address sends a DHCP…

### PR DESCRIPTION
**Pull request by colin-riddell**
A small PR that gives us the ability to know when a specific host is up and making non PXE DHCP requests.

Am trying to touch as little code in this as possible.

### CODE REVIEW

File: pypxe/tftp.py [hunk 1]
```diff
@@ -273,3 +273,4 @@ def listen(self):
             [client.send_block() for client in self.ongoing if client.no_ack()]
             # if we have run out of retries, kill the client
             [client.complete() for client in self.ongoing if client.no_retries()]
+
```
Comment (pdmorrow) [inline pypxe/tftp.py]: random newline.

File: pypxe/dhcp.py [hunk 2]
```diff
@@ -306,6 +317,14 @@ def listen(self):
             self.logger.debug('Parsed received options')
             self.logger.debug('<--BEGIN OPTIONS-->')
             self.logger.debug('{0}'.format(repr(self.leases[client_mac]['options'])))
+
```
Comment (pdmorrow) [inline pypxe/dhcp.py]: I'd add a comment to say what is going on here with all this fiddling with the dhcp options.

**Comment by pdmorrow** [comment]
lgtm.

File: pypxe/dhcp.py [hunk 3]
```diff
@@ -22,7 +22,16 @@ class DHCPD:
         https://en.wikipedia.org/wiki/Dynamic_Host_Configuration_Protocol,
         and http://www.pix.net/software/pxeboot/archive/pxespec.pdf.
     '''
-    def __init__(self, **server_settings):
+    def __init__(self, booted_callback=None, expected_hostname=None, **server_settings):
+        '''
+
+        :param booted_callback: This function is called when the
+        expected_hostname is recognised over DHCP, indicating your host is
+        booted
```
Comment (pdmorrow) [inline pypxe/dhcp.py]: As discussed we need to be careful as this does not mean the os is booted, it means the dhcp client on the module has been assigned an address - which happens during boot.

Comment (colin-riddell) [inline pypxe/dhcp.py]: Agreed. I do further checks on the client.

**Pull request merged**